### PR TITLE
Improve testing and some fixes

### PR DESCRIPTION
**Pull request by dlax**
This adds some test infrastructure so that centos6 installation is actually tested on travis-ci.
This also adds some fixes that had to be done to make tests pass.

### CODE REVIEW

File: run-tests.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        raise RuntimeError("Cannot test if image exists")
+
+
+def _build(image, salt=False):
+    dockerfile = "test/{0}.Dockerfile".format(image)
+    tag = get_tag(image, salt)
+    if salt:
+        dockerfile_content = open(dockerfile, "rb").read()
+        dockerfile_content += (
+            b"\n"
```
Comment (ygversil) [inline run-tests.py:40]: Couldn't this whole string be enclosed within a 3-double-quote delimiter (`"""`) ?

File: run-tests.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            b"ADD test/minion.conf /etc/salt/minion.d/minion.conf\n"
+            b"ADD test/salt /srv/salt\n"
+            b"ADD test/pillar /srv/pillar\n"
+            b"ADD . /srv/formula\n"
+            b"RUN salt-call -l debug state.highstate\n"
+            #b"RUN PATH=/usr/pgsql-9.4/bin/:$PATH salt-call state.highstate\n"
+        ).format(BASEDIR)
```
Comment (ygversil) [inline run-tests.py]: Don't see where `.format()` is needed

File: run-tests.py (hunk 1 above)
Comment (dlax) [inline run-tests.py:40]: Yg Versil a écrit :

> In run-tests.py
> https://github.com/logilab/ckan-formula/pull/6#discussion_r64340699:
> 
> > -    rc = subprocess.call("docker history -q {0} > /dev/null".format(image), shell=True)
> > -    if rc == 0:
> > -        return True
> > -    elif rc == 1:
> > -        return False
> > -    else:
> > -        raise RuntimeError("Cannot test if image exists")
> >   +
> >   +
> >   +def _build(image, salt=False):
> > -    dockerfile = "test/{0}.Dockerfile".format(image)
> > -    tag = get_tag(image, salt)
> > -    if salt:
> > -        dockerfile_content = open(dockerfile, "rb").read()
> > -        dockerfile_content += (
> > -            b"\n"
> 
> Couldn't this whole string be enclosed within a 3-double-quote delimiter
> (|"""|) ?

It could but this would look ugly within the function because of left 
dedent.

BTW, I just copied this code from logilab/saemref-formula.

File: run-tests.py (hunk 2 above)
Comment (dlax) [inline run-tests.py]: Yg Versil a écrit :

> In run-tests.py
> https://github.com/logilab/ckan-formula/pull/6#discussion_r64340878:
> 
> > +
> > +
> > +def _build(image, salt=False):
> > -    dockerfile = "test/{0}.Dockerfile".format(image)
> > -    tag = get_tag(image, salt)
> > -    if salt:
> > -        dockerfile_content = open(dockerfile, "rb").read()
> > -        dockerfile_content += (
> > -            b"\n"
> > -            b"ADD test/minion.conf /etc/salt/minion.d/minion.conf\n"
> > -            b"ADD test/salt /srv/salt\n"
> > -            b"ADD test/pillar /srv/pillar\n"
> > -            b"ADD . /srv/formula\n"
> > -            b"RUN salt-call -l debug state.highstate\n"
> > -            #b"RUN PATH=/usr/pgsql-9.4/bin/:$PATH salt-call state.highstate\n"
> > -        ).format(BASEDIR)
> 
> Don't see where |.format()| is needed

Updated.

File: run-tests.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    dockerfile = "test/{0}.Dockerfile".format(image)
+    tag = get_tag(image, salt)
+    if salt:
+        dockerfile_content = open(dockerfile, "rb").read()
+        dockerfile_content += (
+            b"\n"
+            b"ADD test/minion.conf /etc/salt/minion.d/minion.conf\n"
+            b"ADD test/salt /srv/salt\n"
+            b"ADD test/pillar /srv/pillar\n"
+            b"ADD . /srv/formula\n"
```
Comment (philpep) [inline run-tests.py]: I think you should explicitly ADD wanted directories here (ckan, _states). If not, any change to the current directory (temp file, .git and other unrelated files) will disable the use of the docker cache and trigger (possibly unwanted) rebuilds when using `./run-tests build`.

Comment (dlax) [inline run-tests.py]: Philippe Pepiot a écrit :

> I think you should explicitly ADD wanted directories here (ckan,
> _states). If not any change to the current directory (temp file, .git
> and other unrelated files) will disable the use of the docker cache and
> trigger (possibly unwanted) rebuilds when using |./run-tests build|.

Done in 05507c3bf991ae06a666b0b8167c149d58e902c7.
